### PR TITLE
Add metadata field to low-code schema

### PR DESCRIPTION
**Pull request by lmossman**
## What
Relates to https://github.com/airbytehq/airbyte/issues/25321

We now have a use case where the Connector Builder needs to store some UI-specific data in the connector manifest, as the manifest is the single resource that the builder uses to persist its state.

Specifically, the Connector Builder needs the ability to store an "auto-import schema" boolean flag per stream in the connector manifest. After discussion with Joe and Maxime, we agreed that persisting this in a generic top-level `metadata` field of the connector manifest would be the most straightforward approach, for the following reasons:
- Doesn't require any changes to the persistence layer for connector builder connectors (still just persists the manifest YAML)
- Doesn't affect any existing components in the declarative schema
- Is not specific to the builder - since this is just a generic `metadata` field, technically any consumer of low-code manifests could use this field for whatever they want
- Manifests without any `metadata` set will still be compatible with the Builder - it will just fall back to the default values that it uses when first initializing the metadata values

## 🚨 User Impact 🚨
No breaking changes, since this just adds an optional field to the schema.

### CODE REVIEW

**Review by girarda — APPROVED** [review]
sounds good to me

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ properties:
     type: object
   spec:
     "$ref": "#/definitions/Spec"
+  metadata:
+    type: object
+    description: For internal use only - used by consumers of declarative manifests for storing related metadata
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:32]: what does `internal` mean in this context?

Comment (lmossman) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:32]: I could reword it to say "For internal Airbyte use only" ?

Comment (sherifnada) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:32]: That would be useful! but yeah just asking a question, not really requesting changes

Comment (lmossman) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:32]: Added "Airbyte" to the description and added a note telling people not to modify this field manually

**Review by flash1293 — APPROVED** [review]
LGTM
